src/StatsdClient/StatsdTCP.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+
+namespace StatsdClient
+{
+    public class StatsdTCPClient : IDisposable, IStatsdClient
+    {
+        public IPEndPoint IPEndpoint { get; private set; }
+
+        private readonly Socket _clientSocket;
+        private readonly string _name;
+        private readonly int _port;
+
+        public StatsdTCPClient(string name, int port = 8125)
+        {
+            _name = name;
+            _port = port;
+
+            _clientSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+            var ipAddress = GetIpv4Address(name);
+            IPEndpoint = new IPEndPoint(ipAddress, _port);
+        }
+
+        private IPAddress GetIpv4Address(string name)
```

Comment:
Maybe pull this out to a separate class so both TCP & UDP implementations can share it?